src/api/cli/integration/integration_test.go

```diff
@@ -1341,3 +1357,236 @@ func TestLiveWalletDir(t *testing.T) {
 
 	require.Equal(t, walletDir, strings.Trim(string(output), "\n"))
 }
+
+// TestLiveSend sends coin from specific wallet file, user should manually specify the
+// wallet file by setting the enviroment variables: WALLET_DIR and WALLET_NAME. The WALLET_DIR
+// points to the directory of the wallet, and WALLET_NAME represents the wallet file name.
+//
+// Note:
+// 1. This test might modify the wallet file, in order to avoid losing coins, we don't send coins to
+// addresses that are not belong to the wallet, when addresses in the wallet are not sufficient, we
+// will automatically generate enough addresses as coin recipient.
+// 2. The wallet should have at least 2 skycoins, and at least 0.5 coin in the first
+// address of the wallet for testing -a(from address) option.
+func TestLiveSend(t *testing.T) {
+	if !doLive(t) {
+		return
+	}
+
+	w := prepareAndCheckWallet(t, 4)
+
+	tt := []struct {
+		name    string
+		args    func() []string
+		checkTx func(t *testing.T, txid string)
+	}{
+		{
+			"send with -a(from address) flag",
+			func() []string {
+				// Set the third address as the receiver address
+				return []string{"send", "-a", w.Entries[0].Address.String(), w.Entries[1].Address.String(), "0.5"}
+			},
+			func(t *testing.T, txid string) {},
+		},
+		{
+			"send to multiple address",
+			func() []string {
+				addrCoins := []struct {
+					Addr  string `json:"addr"`
+					Coins string `json:"coins"`
+				}{
+					{
+						w.Entries[1].Address.String(),
+						"0.5",
+					},
+					{
+						w.Entries[2].Address.String(),
+						"0.5",
+					},
+				}
+
+				v, err := json.Marshal(addrCoins)
+				require.NoError(t, err)
+
+				return []string{"send", "-m", string(v)}
+			},
+			func(t *testing.T, txid string) {
+				tx := getTransaction(t, txid)
+				// get the total input hours the transaction has.
+				_, hours := getUxoutCoinsAndHours(t, tx.Transaction.Transaction.In)
+				// Only when hours >= 4, the two recipient addresses could all receive 1 coinhours
+				if hours >= 4 {
+					// Confirms the second address has 0.5 coin and 1 coinhour
+					checkCoinsAndCoinhours(t, tx, w.Entries[1].Address.String(), "0.500000", 1)
+					// Confirms the third address has 1 coin and 1 coinhour
+					checkCoinsAndCoinhours(t, tx, w.Entries[2].Address.String(), "0.500000", 1)
+				}
+
+				// If the total input coin hours is <= 1, the two recipient addresses should receive no coinhours.
+				if hours <= 1 {
```

Comment:
I think the test should always require a certain number of coinhours, so that the test does not have different behavior depending on the number of coinhours.  We can create spends to control the number of coinhours in another address to test what happens with 1 coinhour and 0 coinhours.